pkg/provider/cloud/vsphere/tags.go

```diff
@@ -36,6 +38,15 @@ func reconcileTags(ctx context.Context, restSession *RESTSession, cluster *kuber
 		return nil, fmt.Errorf("failed to sync deleted tags %w", err)
 	}
 
+	cluster, err := update(ctx, cluster.Name, func(cluster *kubermaticv1.Cluster) {
+		if !kuberneteshelper.HasFinalizer(cluster, tagCleanupFinalizer) {
+			kuberneteshelper.AddFinalizer(cluster, tagCleanupFinalizer)
+		}
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to add finalizer %s on vsphere cluster object: %w", tagCleanupFinalizer, err)
+	}
```

Comment:
This finalizer is already removed in the `provider.go` file during cleanup, no? Am I missing anything?

https://github.com/kubermatic/kubermatic/blob/1c585277eb00e03e6b8ac6660497154aef78e2dd/pkg/provider/cloud/vsphere/provider.go#L229-L236

Reply:
Stupid me, yeah this is _adding_ the finalizer.